eppo_core/src/events/vec_event_queue.rs

```diff
@@ -0,0 +1,197 @@
+use crate::events::queued_event::{QueuedEvent, QueuedEventStatus};
+use std::collections::VecDeque;
+use std::sync::{Arc, Mutex};
+
+pub trait EventQueue {
+    /// Pushes an event to the end of the queue.
+    fn push(&self, event: QueuedEvent) -> Result<(), QueueError>;
+
+    /// Returns the next batch of events with the provided status.
+    /// and with status PENDING. Events are returned in FIFO order.
+    fn next_batch(&self, status: QueuedEventStatus) -> Vec<QueuedEvent>;
+
+    /// Returns whether the queue contains enough events for delivering *at least* one batch.
+    fn is_batch_full(&self) -> bool;
+}
```

Comment:
@rasendubi I'm aware this might conflict with your work, but we can resolve conflicts if needed, I don't expect the trait implementation will diverge too much. What do you think?

Reply:
It is going to be quite different to account for usage from multiple threads and to avoid re-notification of the delivery task.

I'm in the design phase though, so don't have much code to cause conflict. So let's merge this PR and I will base my work on top